### PR TITLE
Update ruby-saml gem to latest

### PR DESCRIPTION
**Pull request by omgitsbillryan**
Ran 
```
gem update --conservative ruby-saml
```  

`ruby-saml` 1.5.0 V.S. 1.3.1 had some breaking changes that I had to address.

### CODE REVIEW

File: spec/factories/rubysaml_settings.rb [hunk 1]
```diff
@@ -12,11 +12,6 @@
     idp_slo_target_url              'https://api.idmelabs.com/saml/SingleLogoutService'
     idp_sso_target_url              'https://api.idmelabs.com/saml/SingleSignOnService'
     name_identifier_format          'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified'
-
-    trait :invalid_cert do
-      # fun fact - this is an actual cert ID.me once served to us in prod (it contains \r chars)
-      idp_cert File.read("#{::Rails.root}/spec/fixtures/files/invalid_idme_cert.crt")
-    end
```
Comment (omgitsbillryan) [inline spec/factories/rubysaml_settings.rb]: Not used anymore. Replaced with VCR.

File: lib/saml/settings_service.rb [hunk 2]
```diff
@@ -28,10 +28,14 @@ def saml_settings
       end
 
       def merged_saml_settings
-        OneLogin::RubySaml::IdpMetadataParser.new.parse(metadata, settings: settings)
-      rescue => e
-        log_message_to_sentry("SAML::SettingService failed to parse SAML metadata: #{e.message}", :error)
```
Comment (omgitsbillryan) [inline lib/saml/settings_service.rb]: When `metadata` was `nil`, this used to just return the `settings` that we passed in.  In `ruby-saml 1.5.0`, an `ArgumentError` is raised if `metadata` does not contain an `IDPSSODescriptor ` xml element ([Source](https://github.com/onelogin/ruby-saml/blob/e9d454b5bf1d203c53fd008a7c2200aff9e4f7d6/lib/onelogin/ruby-saml/idp_metadata_parser.rb#L106)).

I made this change to tests pass but I'm actually debating if raising might actually be better...

Comment (saneshark) [reply lib/saml/settings_service.rb]: I would think raising makes sense, this would happen if ID.me is down for some reason right?

Comment (omgitsbillryan) [reply lib/saml/settings_service.rb]: yes.  in the past the metadata endpoint has been offline and we got a 404 `text/html` response with an ID.me 404 page.

**Comment by omgitsbillryan** [comment]
Ya, I'd like at least half a day to perform manual testing in `dev` and or `staging`.  Login on the review instances is not quite the same as on `dev` & `staging` so testing on that is _not quite_ what we want.

**Comment by omgitsbillryan** [comment]
@saneshark the only way I know to get stuff into `staging` is to merge the PR to `master`.  So perhaps I should merge this 3pm EST on monday and test monday evening and Tuesday morning?

**Comment by patrickvinograd** [comment]
Anybody can log in to jenkins and deploy any branch to staging.

**Comment by patrickvinograd** [comment]
Your changes are liable to get overwritten by a later merge to master, but you can just keep jenkins open in a tab and kill any staging deploys if you're not done testing

**Pull request merged**